quinn-proto/src/packet.rs

```diff
@@ -7,21 +7,50 @@ use slog;
 use coding::{self, BufExt, BufMutExt};
 use {MAX_CID_SIZE, VERSION};
 
+pub struct Packet {
+    pub header: Header,
+    pub header_data: Bytes,
+    pub payload: BytesMut,
+}
+
+impl Packet {
+    pub fn decode(
+        mut packet: BytesMut,
+        dest_id_len: usize,
+    ) -> Result<(Self, BytesMut), HeaderError> {
+        let (header_len, payload_len, header) = Header::decode(&mut packet, dest_id_len)?;
+        let header_data = packet.split_to(header_len).freeze();
+        let payload = packet.split_to(payload_len);
+        Ok((
+            Packet {
+                header,
+                header_data,
+                payload,
+            },
+            packet,
+        ))
+    }
+}
+
 #[derive(Debug, Clone)]
 pub enum Header {
     Long {
-        ty: u8,
+        ty: LongType,
         source_id: ConnectionId,
         destination_id: ConnectionId,
         number: u32,
     },
+    Retry {
+        source_id: ConnectionId,
+        destination_id: ConnectionId,
+        orig_dst_cid: ConnectionId,
```

Comment:
Need retry token

Reply:
Yeah, I was going to put that in the payload.